### PR TITLE
VL-100_Datepickers-&-Checkbox-useAttrs-refactoring_Vitalii-Dudnik

### PR DESCRIPTION
**Pull request by KinduD21**
https://ilevel.atlassian.net/jira/software/c/projects/VL/boards/11?selectedIssue=VL-100

### CODE REVIEW

File: src/ui.form-calendar/useAttrs.js [hunk 1]
```diff
@@ -1,375 +1,176 @@
...
+      extend: computed(() => [
+        extendingKeysClasses.day.value,
+        extendingKeysClasses.currentDate.value,
+      ]),
+    },
+    currentDayInRange: {
+      extend: computed(() => [
+        extendingKeysClasses.day.value,
+        extendingKeysClasses.dateInRange.value,
+        extendingKeysClasses.currentDate.value,
```
Comment (itsJohnnyGrid) [inline src/ui.form-calendar/useAttrs.js:70]: here all classes should be in base

File: src/ui.form-calendar/useAttrs.js [hunk 2]
```diff
@@ -18,12 +20,13 @@ export default function useAttrs(props) {
     "currentDate",
     "edgeDateInRange",
     "dateInRange",
-    "day",
     "timepickerInput",
   ];
+
   const extendingKeysClasses = getExtendingKeysClasses(extendingKeys);
```
Comment (itsJohnnyGrid) [inline src/ui.form-calendar/useAttrs.js:26]: `const extendingKeysClasses = getExtendingKeysClasses([...extendingKeys, "day"]);`

File: src/ui.form-calendar/useAttrs.js [hunk 3]
```diff
@@ -1,375 +1,179 @@
...
+    },
+    yearInRange: {
+      base: computed(() => [extendingKeysClasses.dateInRange.value]),
+    },
+    singleYearInRange: {
+      base: computed(() => [extendingKeysClasses.dateInRange.value]),
+    },
+    currentYearInRange: {
+      extend: computed(() => [
+        extendingKeysClasses.currentDate.value,
```
Comment (itsJohnnyGrid) [inline src/ui.form-calendar/useAttrs.js:150]: It looks like the logic is wrong:
image

Double check all configs again. Here is non typical logic (was applied before), be aware to it.

Comment (KinduD21) [reply src/ui.form-calendar/useAttrs.js:172]: I've checked and changed extendingKeys strategy from `extend` to `base`, add separate `"month"` and `"year" `keys to `getExtendingKeysClasses` function

**Review by itsJohnnyGrid — APPROVED** [review]

**Pull request merged**